adyenv6b2ccheckoutaddon/acceleratoraddon/web/src/com/adyen/v6/controllers/pages/AdyenSummaryCheckoutStepController.java

```diff
@@ -178,6 +183,21 @@ public String placeOrder(@ModelAttribute("placeOrderForm") final PlaceOrderForm
             } catch (Exception e) {
                 LOGGER.error(ExceptionUtils.getStackTrace(e));
             }
+        } else if(PAYMENT_METHOD_POS.equals(adyenPaymentMethod)) {
+            try {
+                OrderData orderData = adyenCheckoutFacade.initiatePosPayment(cartData);
+                LOGGER.debug("Redirecting to confirmation!");
+                return redirectToOrderConfirmationPage(orderData);
+            } catch (ApiException e) {
+                LOGGER.error("API exception " + e.getError(), e);
+            } catch (AdyenNonAuthorizedPaymentException e) {
```

Comment:
Is AdyenNonAuthorizedPaymentException exception thrown in this payment method ?

Reply:
It's thrown for ResultType.FAILURE, following a similar flow to REFUSED on the checkout authorization method.